### PR TITLE
Finished IPFS support, added improvements

### LINKED ISSUE #12: Introducing `webrunify`
Not really an issue, but something that might be of interest.

I've created a browserify plugin (and CLI program) called [webrunify](https://github.com/brechtcs/webrunify) to bundle up CommonJS modules written for Node and expose them as ES modules with all their dependencies included. These bundles can then be shared as standalone modules and imported into Webrun or Beaker just like any other ES module:

```js
import lib from 'dat://example.com/lib.bundle.js'
```
I think this could be useful for people who want to use modules from NPM, but prefer to move away from `--allow-require` (as you rightly recommend in the docs).

### LINKED ISSUE #16: env: node\r: No such file or directory
I get following error on osx:

```
env: node\r: No such file or directory
```

I believe it's due to line brakes used in file

### LINKED ISSUE #18: require works only if webrun is installed as local dependency
Forking this from #16 so issue isn't derailed to this.

> Oh wow, I think it's because I'm a dingus.
> 
> https://github.com/RangerMauve/webrun/blob/master/src/plugins/node.js#L67
> 
> Basically, it uses the require logic relative to where webrun is installed rather than the present working directory. Not 100% sure what to do about that unless there's a module that lets you require relative to an arbitrary path.

### PR DESCRIPTION
**Pull request by RangerMauve**
- Added support for loading `ipfs://` and `ipns://` URLs
- Added `ipfs` global which is an instance of `js-ipfs`
- Fixes #18 by using `require.resolve` with `process.cwd()`
- Added WhatWG Streams to globals (ReadableStream, WritableStream, etc)
- Closes #12 by mentioning `webrunify` in the README
- Fixed an issue with dynamic `import()` support
- Fixes #16 with a prepublish script which changes the windows file endings to unix file endings.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,11 +26,18 @@ webrun ./example.js
...
 
 ```javascript
+// Load code from an HTTPS server
 import example from "https://rangermauve.hashbase.io/esm.js";
-import p2pexample from "dat://rangermauve.hashbase.io/esm.js";
+
+// Load from the dat network
+import datExample from "dat://rangermauve.hashbase.io/esm.js";
+
+// Load from the IPFS network. Might not always be online.
```
Comment (brechtcs) [inline README.md]: Why the "Might not always be online" warning here, and not for `dat`?

(I'm guessing it's that the IPFS ecosystem doesn't have a pinning service like Hashbase, but don't know enough about it to be sure.)

File: src/plugins/ipfs.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    const path = url.pathname
+
+    const hostname = url.hostname
+
+    const hash = await ipfs.name.resolve(`ipns/${hostname}`)
+
+    const finalURL = `${hash}${path}`
+
+    const result = await ipfs.get(finalURL)
```
Comment (brechtcs) [inline src/plugins/ipfs.js]: This and next 4 lines are duplicated in `getIPFSFile`

**Comment by brechtcs** [comment]
Not essential to the PR, but I noticed the commits aren't linked to your Github profile as they should. I guess you made them on a different device with a different email in `gitconfig`.

Coincidentally, I just fixed this myself somewhere else today. So in case you need it, this is the script I used:

```sh
git filter-branch -f --commit-filter '
if [ "$GIT_AUTHOR_NAME" = "Brecht Savelkoul" ];
then
    GIT_AUTHOR_EMAIL="brecht.savelkoul@gmail.com";
    GIT_COMMITTER_EMAIL="brecht.savelkoul@gmail.com";
    git commit-tree "$@";
else
    git commit-tree "$@";
fi' HEAD

git push -f
```

File: README.md (hunk 1 above)
Comment (RangerMauve) [reply README.md]: I don't have the contents pinned personally, but if you know of a pinning service for IPFS I'll definitely add it there

File: src/plugins/ipfs.js (hunk 2 above)
Comment (RangerMauve) [reply src/plugins/ipfs.js]: Yeah, good point I'll refactor it into a single function.

**Comment by RangerMauve** [comment]
I was going to do a `squash` merge through GH so I'd still be on the commits.

It's weird that IPFS doesn't have a hashbase equivalent.  ¯\_(ツ)_/¯

@brechtcs Would you mind doing another quick review of the IPFS loading code?

**Comment by RangerMauve** [comment]
Thank you for the review!

**Pull request merged**